cmd/argocd/commands/headless/headless.go

```diff
@@ -115,14 +116,27 @@ type forwardRepoClientset struct {
 	repoClientset  repoapiclient.Clientset
 	err            error
 	repoServerName string
+	kubeClientset  kubernetes.Interface
 }
 
 func (c *forwardRepoClientset) NewRepoServerClient() (io.Closer, repoapiclient.RepoServerServiceClient, error) {
 	c.init.Do(func() {
 		overrides := clientcmd.ConfigOverrides{
 			CurrentContext: c.context,
 		}
-		repoServerPodLabelSelector := common.LabelKeyAppName + "=" + c.repoServerName
+		repoServerName := c.repoServerName
+		repoServererviceLabelSelector := common.LabelKeyComponentRepoServer + "=" + common.LabelValueComponentRepoServer
+		repoServerServices, err := c.kubeClientset.CoreV1().Services(c.namespace).List(context.Background(), v1.ListOptions{LabelSelector: repoServererviceLabelSelector})
```

Comment:
Instead of deriving the argocd instance name from the services, can we not apply the label `app.kubernetes.io/component=repo-server` to the `spec.template.metadata` section of the repo server deployment and then use this label to get the right repo server. 
https://github.com/argoproj/argo-cd/blob/f87897c53c6f04426953f5d3ca781d3240186a60/manifests/base/repo-server/argocd-repo-server-deployment.yaml#L16C9-L16C51

Reply:
Hi @anandf,  I will update the implementation and add the labels "app.kubernetes.io/component=repo-server" to repo-server replica pods in the manifest

Reply:
Switching to `app.kubernetes.io/component=repo-server` component based label might cause confusion when customer provides argument `--repo-server-name` with the name of the argocd instance `argocd-repo-server`. So let's keep the logic of deriving the argocd repo server name from the service name.